node/router/messages/src/lib.rs

```diff
@@ -152,6 +152,28 @@ impl<N: Network> Message<N> {
             Self::UnconfirmedTransaction(..) => 12,
         }
     }
+
+    /// Checks the message byte length. To be used before deserialization.
+    pub fn check_size(bytes: &[u8]) -> io::Result<()> {
+        // Store the length to be checked against the max message size for each variant.
+        let len = bytes.len();
+        if len < 2 {
+            return Err(io::Error::new(io::ErrorKind::InvalidData, "invalid message"));
+        }
+
+        // Check the first two bytes for the message ID.
+        let id_bytes: [u8; 2] = (&bytes[..2])
+            .try_into()
+            .map_err(|_| io::Error::new(io::ErrorKind::InvalidData, "id couldn't be deserialized"))?;
+        let id = u16::from_le_bytes(id_bytes);
+
+        // SPECIAL CASE: check the transaction message isn't too large.
+        if id == 12 && len > N::MAX_TRANSACTION_SIZE {
```

Comment:
@d0cd can this `check_size` method be moved in to the respective file for message type 12?

Reply:
I've moved it down into `UnconfirmedTransaction`, I think that's what you meant? 
One thing to bear in ming is this function takes in _all_ message bytes, not just `UnconfirmedTransaction`s, even though it only does the length check for the latter, it could be expanded in future to check other message types.